pkg/osquery/table/app_icons_darwin.go

```diff
@@ -0,0 +1,77 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa
+#import <Appkit/AppKit.h>
+void Icon(CFDataRef *iconDataRef, char* path) {
+	NSString *appPath = [[NSString stringWithUTF8String:path] stringByStandardizingPath];
+	NSImage *img = [[NSWorkspace sharedWorkspace] iconForFile:appPath];
```

Comment:
I'm so happy you got this direction working

Reply:
Same. It worked great and handles the case when an icon is not defined by the bundle